lib/puppet/network/http/api/master/v3/environment.rb

```diff
@@ -20,14 +20,13 @@ def call(request, response)
     applications.each do |app|
       app_components = {}
       # Turn the 'nodes' hash into a map component ref => node name
-      node_mapping = app['nodes'].map do |k,v|
-        k.type == "Node" ? [k, v] : [v, k]
-      end.reduce({}) do |map, (node, comp)|
-        unless map[comp.ref].nil?
-          raise Puppet::ParseError, "Application #{app} maps component #{comp} to mutiple nodes"
+      node_mapping = {}
+      app['nodes'].each do |node, comps|
+        comps = [comps] unless comp.is_a?(Array)
```

Comment:
ok, will merge - we can always do that later if people use the same pattern here as for other similar things - i.e. that they want a kind of set operation  - "this set + that set". No big deal.